File: src/main/scala/committee/nova/skillful/event/impl/SkillLevelEvent.scala
package committee.nova.skillful.event.impl

import committee.nova.skillful.impl.skill.instance.SkillInstance
import net.minecraft.entity.player.ServerPlayerEntity

object SkillLevelEvent {
  class Up(player: ServerPlayerEntity, skill: SkillInstance, currentLevel: Int) extends SkillLevelEvent(player, skill, currentLevel)

  class Down(player: ServerPlayerEntity, skill: SkillInstance, currentLevel: Int) extends SkillLevelEvent(player, skill, currentLevel) {
    override def isUp: Boolean = false
  }
}

class SkillLevelEvent(private val player: ServerPlayerEntity, private val skill: SkillInstance, private val currentLevel: Int) extends SkillEvent(player, skill) {
  def getCurrentLevel: Int = currentLevel

  def isUp: Boolean = true

  def isDown: Boolean = !isUp
}

File: src/main/java/committee/nova/skillful/common/event/impl/SkillLevelEvent.java
package committee.nova.skillful.common.event.impl;


import committee.nova.skillful.common.skill.SkillInstance;
import net.minecraft.world.entity.player.Player;

public class SkillLevelEvent extends SkillEvent {
    private final long oldLevel;
    private final long newLevel;

    public SkillLevelEvent(Player player, SkillInstance skill, long oldLevel, long newLevel) {
        super(player, skill);
        this.oldLevel = oldLevel;
        this.newLevel = newLevel;
    }

    public long getOldLevel() {
        return oldLevel;
    }

    public long getNewLevel() {
        return newLevel;
    }

    public boolean isUp() {
        return newLevel > oldLevel;
    }

    public boolean isDown() {
        return newLevel < oldLevel;
    }
}
